hnn_core/cell.py

```diff
@@ -137,26 +155,148 @@ def gid(self, gid):
         else:
             raise RuntimeError('Global ID for this cell already assigned!')
 
-    @abstractmethod
-    def get_sections(self):
-        """Get sections in a cell."""
-        pass
-
-    def create_soma(self):
-        """Create soma and set geometry."""
-        # make L_soma and diam_soma elements of self
-        # Used in shape_change() b/c func clobbers self.soma.L, self.soma.diam
-        soma_props = self.soma_props
-
-        self.L = soma_props['L']
-        self.diam = soma_props['diam']
-        self.pos = soma_props['pos']
+    def _set_biophysics(self, p_secs):
+        "Set the biophysics for the default Pyramidal cell."
+
+        # neuron syntax is used to set values for mechanisms
+        # sec.gbar_mech = x sets value of gbar for mech to x for all segs
+        # in a section. This method is significantly faster than using
+        # a for loop to iterate over all segments to set mech values
+
+        # Value depends on distance from the soma. Soma is set as
```

Comment:
"IF the value depends on distance ..."

This is only true some some mechanisms (though it will probably be the rule going forward.) The code below is opaque, but this is due primarily to HOC objects being so darn hard to introspect.